Docs & tests for rejectWithValue (#2)
phryneas/redux-toolkit: docs/usage/usage-with-typescript.md

    // Return the known error for future handling
    return thunkApi.rejectWithValue((await response.json()) as MyKnownError)
  }
  return (await response.json()) as MyData
})
```

While this notation for `state`, `dispatch`, `extra` and `rejectValue` might seem uncommon at first, it allows you to provide only the types for these you actually need - so for example, if you are not accessing `getState` within your `payloadCreator`, there is no need to provide a type for `state`. The same can be said about `rejectValue` - if you don't need to access any potential error payload, you can ignore it.

In addition, you can leverage checks against `action.payload` and `match` as provided by `createAction` as a type-guard for when you want to access known properties on defined types. Example:

- In a reducer

```ts
const usersSlice = createSlice({
  name: 'users',
  initialState: {
    entities: {},
    error: null
  },
  reducers: {},
  extraReducers: builder => {
    builder.addCase(updateUserById.fulfilled, (state, { payload }) => {
      state.entities[payload.id] = payload
    })
    builder.addCase(updateUserById.rejected, (state, action) => {
      if (action.payload) {

>         // Being that we passed in MyKnownError to rejectType in `updateUserById`, the type information will be available here.

Reviewer: ```suggestion
        // Since we passed in `MyKnownError` to `rejectType` in `updateUserById`, the type information will be available here.
```

        state.error = action.payload.errorMessage
      } else {
        state.error = action.error
      }
    })
  }
})
```

- In a component

```ts
const updateUser = async userData => {
  const { id, ...userFields } = userData
  const resultAction = await dispatch(updateUserById(userFields))
  if (updateUser.fulfilled.match(resultAction)) {
    const user = unwrapResult(resultAction)
    showToast('success', `Updated ${user.name}`)
  } else {
    if (resultAction.payload) {

>       // Being that we passed in MyKnownError to rejectType in `updateUserById`, the type information will be available here.

Reviewer: ```suggestion
      // Since we passed in `MyKnownError` to `rejectType` in `updateUserById`, the type information will be available here.
```
